### PR TITLE
Sign Up test

### PR DESCRIPTION
**Pull request by starikovskii**

### CODE REVIEW

File: app/src/test/java/com/client/shop/ui/account/contract/SignUpPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@Suppress("FunctionName")
```
Comment (DenysZP) [inline app/src/test/java/com/client/shop/ui/account/contract/SignUpPresenterTest.kt]: Please, remove redundant annotation.

File: domain/src/test/java/com/domain/interactor/account/SignUpUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import com.domain.repository.AuthRepository
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.robolectric.annotation.Config
+
+@Suppress("FunctionName")
```
Comment (DenysZP) [inline domain/src/test/java/com/domain/interactor/account/SignUpUseCaseTest.kt]: Please, remove redundant annotation.

File: app/src/test/java/com/client/shop/ui/account/contract/SignUpPresenterTest.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    @Test
+    fun shouldExecuteUseCaseOnValidData() {
+        given(useCase.buildUseCaseCompletable(any())).willReturn(Completable.complete())
+        presenter.signUp("", "", correctEmail, correctPass, "")
+        verify(useCase).execute(any(), any(), eq(SignUpUseCase.Params("", "", correctEmail, correctPass, "")))
+    }
+
+    @Test
+    fun shouldShowErrorOnInvalidPass() {
+        presenter.signUp("", "", "", incorrectPass, "")
```
Comment (plzen) [inline app/src/test/java/com/client/shop/ui/account/contract/SignUpPresenterTest.kt]: You should mock validator and check is presenter calls proper method

File: app/src/test/java/com/client/shop/ui/account/contract/ChangePasswordPresenterTest.kt [hunk 4]
```diff
@@ -117,8 +113,8 @@ class ChangePasswordPresenterTest {
     fun shouldShowPasswordMatchErrorOnDifferentPass() {
         presenter.changePassword("12345678", "123456789")
         verify(view).passwordsMatchError()
-        verify(useCase, never()).execute(any(), any(), eq("12345678"))
-        verify(useCase, never()).execute(any(), any(), eq("123456789"))
+        verify(useCase, never()).execute(any(), any(), any())
```
Comment (plzen) [inline app/src/test/java/com/client/shop/ui/account/contract/ChangePasswordPresenterTest.kt]: Do we need to verify this twice?

Comment (starikovskii) [reply app/src/test/java/com/client/shop/ui/account/contract/ChangePasswordPresenterTest.kt]: no, it is a duplicate code, will delete it

**Pull request merged**